livox_ros2_driver/livox_ros2_driver/lddc.cpp

```diff
@@ -712,433 +702,403 @@ void Lddc::initializeDiagnostics()
   cur_node_->get_node_timers_interface()->add_timer(timer_, nullptr);
 }
 
+void Lddc::registerDiagnosticsUpdater(const std::string &broadcast_code)
+{
+  updater_.add(
+      fmt::format("livox_temperature-{}", broadcast_code),
+      std::bind(
+          &Lddc::checkTemperature, this, std::placeholders::_1,
+          broadcast_code));
+
+  updater_.add(
+      fmt::format("livox_internal_voltage-{}", broadcast_code),
+      std::bind(
+          &Lddc::checkVoltage, this, std::placeholders::_1,
+          broadcast_code));
+
+  updater_.add(
+      fmt::format("livox_motor_status-{}", broadcast_code),
+      std::bind(
+          &Lddc::checkMotor, this, std::placeholders::_1,
+          broadcast_code));
+
+  updater_.add(
+      fmt::format("livox_optical_window-{}", broadcast_code),
+      std::bind(
+          &Lddc::checkDirty, this, std::placeholders::_1,
+          broadcast_code));
+
+  updater_.add(
+      fmt::format("livox_firmware_status-{}", broadcast_code),
+      std::bind(
+          &Lddc::checkFirmware, this, std::placeholders::_1,
+          broadcast_code));
+
+  if (check_pps_signal_) {
+    updater_.add(
+        fmt::format("livox_pps_signal-{}", broadcast_code),
+        std::bind(
+            &Lddc::checkPPSSignal, this, std::placeholders::_1,
+            broadcast_code));
+  }
```

Comment:
[Question] Why do you have to choose whether to add pps_signal or not here? Can we not filter it in diagnostic_aggregator or autoware_error_monitor?

Reply:
@kenji-miyake 

> [Question] Why do you have to choose whether to add pps_signal or not here? Can we not filter it in diagnostic_aggregator or autoware_error_monitor?

It's not me who added pps_signal param. Will we remove this param?

Reply:
> It's not me who added pps_signal param. Will we remove this param?

@kosuke55 hmm, then let's leave it as it is in this PR.

Reply:
It was from https://github.com/tier4/livox_ros2_driver/pull/2/commits/630ee09cec19bd7b5976fce3572c41ff11d41cc7.